doc/api/fs.md

```diff
@@ -1381,6 +1381,26 @@ the specified file descriptor. This means that no `'open'` event will be
 emitted. Note that `fd` should be blocking; non-blocking `fd`s should be passed
 to [`net.Socket`][].
 
+Also note that the blocking fd, if pointing to a character device (such as
+keyboard or sound card) can potentially block the main thread on stream close.
+This is because these devices do not produce EOF character as part of their data
+flow cycle, and thereby exemplifies endless streams. As a result, they do not
+respond to `stream.close()` call. A workaround is to close the stream first
+using `stream.close()` and then push a random character into the stream.
+This unblocks the reader thread, leads to the completion of the data flow cycle,
+and the actual closing of the stream.
+
+```js
+const fs = require('fs');
+// create a stream from some character  device
```

Comment:
Nit: it seems we usually start comments with an uppercase letter and end them with a period.